ecl/agentexec/agentexec.cpp

```diff
@@ -206,16 +212,18 @@ int CEclAgentExecutionServer::run()
 
 //---------------------------------------------------------------------------------
 
-#ifdef _CONTAINERIZED
+// NB: WaitThread only used by if pool created (see CEclAgentExecutionServer ctor)
 class WaitThread : public CInterfaceOf<IPooledThread>
 {
 public:
     WaitThread(const char *_dali, const char *_apptype, const char *_queue) : dali(_dali), apptype(_apptype), queue(_queue)
     {
+        isThorAgent = streq("thor", apptype);
     }
     virtual void init(void *param) override
     {
-        wuid.set((const char *) param);
+        auto context = *static_cast<std::tuple<unsigned, const char *, const char *, const char *> *>(param);
```

Comment:
> slightly cleaner if a typedef was used for `std::tuple<unsigned, const char *, const char *, const char *>`

agree, will use a typedef.

> Will this cause the tuple object to be cloned? Rather than using *context in the next line.

Yes, it should have been declared as: auto &context. I'll change it.